File: solutions/c/collatz-conjecture/1/collatz_conjecture.c
#include "collatz_conjecture.h"

int steps(int start) {
    if (start <= 0) return ERROR_VALUE;
    unsigned n = 0;
    unsigned x = start;
    while (x != 1) {
        x = x % 2 ? x * 3 + 1 : x / 2;
        ++n;
    }
    return n;
}

File: solutions/go/collatz-conjecture/1/collatz_conjecture.go
package collatzconjecture

import "errors"

func CollatzConjecture(n int) (steps int, err error) {
	if n < 1 {
		return 0, errors.New("n < 1")
	}
	for n > 1 {
		steps++
		if n%2 == 0 {
			n = n / 2
		} else {
			n = n*3 + 1
		}
	}
	return
}
